webrtc/interfaces.html

```diff
@@ -0,0 +1,74 @@
+<!doctype html>
+<html>
+  <head>
+    <title>"WebRTC IDL Tests</title>
+  </head>
+  <body>
+    <script src=/resources/testharness.js></script>
+    <script src=/resources/testharnessreport.js></script>
+    <script src=/resources/WebIDLParser.js></script>
+    <script src=/resources/idlharness.js></script>
+    <script>
+      'use strict';
+
+      function initRTCCertificate(idlArray) {
+        return RTCPeerConnection.generateCertificate({
+          name: 'RSASSA-PKCS1-v1_5',
+          modulusLength: 2048,
+          publicExponent: new Uint8Array([1, 0, 1]),
+          hash: 'SHA-256'
+        })
+        .then(cert => {
+          window.rtcCert = cert;
+          idlArray.add_objects({'RTCCertificate': ['rtcCert']});
+         });
+      }
+
+      // A customized version of test_harness that returns
+      // promise of the test completing as well as allow
+      // other promise tests to run in parallel.
+      function promise_test(testFunc, testName, properties) {
```

Comment:
I see, you don't want the whole test to fail if `initRTCCertificate()` does, and apparently Edge doesn't have `RTCPeerConnection.generateCertificate()`. How about just adding `if (!RTCPeerConnection.generateCertificate) return null` and then not adding the instance if it's null?